src/edit.go

```diff
@@ -46,16 +53,34 @@ func (c *EditCommand) Run(args []string) int {
 
 	WriteSecretToFile(file, secret.Data)
 
-	err = EditFile(file.Name())
-	if err != nil {
-		c.Ui.Error(fmt.Sprintf("Unable to edit secret file %q", err))
-		return 1
-	}
+	var data map[string]interface{}
+	secret_is_valid := false
 
-	data, err := ParseSecret(file.Name())
-	if err != nil {
-		c.Ui.Error(fmt.Sprintf("Secret has not changed %q", err))
-		return 1
+	// Re-open the text editor if the parsing of the resulting secret fails
+	for(secret_is_valid == false) {
+
+		secret_is_valid = true
+
+		err = EditFile(file.Name())
+		if err != nil {
+			c.Ui.Error(fmt.Sprintf("Unable to edit secret file %q", err))
+			return 1
+		}
+
+		data, err = ParseSecret(file.Name())
+		switch err {
+		case ErrDuplicateKey:
```

Comment:
To avoid duplicate code you can list cases comma separated.
case ErrDuplicateKey, ErrMultipleDelimiters, ErrMissingDelimiter:

Reply:
Thanks! I tried to lookup on how to "or"-connect the errors into one statement. However I could not figure it out.  That's exactly what I was looking for.